jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SamProjectWizard.kt

```diff
@@ -88,6 +88,9 @@ abstract class SamProjectTemplate {
         }
     }
 
+    // defined so that we can restore the template selection when the runtime selection changes
+    override fun equals(other: Any?) = other?.let { this::class.java == it::class.java } ?: super.equals(other)
```

Comment:
The equals check feels odd, is it really needed?

We only create 1 SDK template ever and store them in lists, so why is the default identity check not working?

Reply:
when setting the selected item with `templateComboBox.selectedItem = selectedTemplate`, a equality check runs against everything in the model to figure out which one

Reply:
to be more specific, the specific instances in the model change after a runtime change:

https://github.com/aws/aws-toolkit-jetbrains/blob/a110ba448825fb7b3bff298d3e19bebad1be807d/jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SamInitSelectionPanel.kt#L117-L124

Reply:
swapped to be keyed off `displayName()` instead and added hashCode() override